apps/omg_watcher/lib/omg_watcher/block_getter/status.ex

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.BlockGetter.Status do
+  @moduledoc """
+  Keeps track and exposes the current status of `OMG.Watcher.BlockGetter`.
+
+  The reason to have this is that the parent `OMG.Watcher.BlockGetter` is doing a lot of synchronous heavy lifting and
+  can easily become unresponsive when syncing (especially when catching up a lot of blocks).
+
+  Expects current status to be eagerly pushed to it.
+  **NOTE**: The API to _get_ the status remained in `OMG.Watcher.BlockGetter` for clarity
+  """
+
+  use GenServer
+
+  alias OMG.Watcher.BlockGetter.Core
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  @doc """
+  Overwrites the currently stored status with the provided one
+  """
+  @spec update(Core.chain_ok_response_t()) :: :ok
+  def update(status), do: GenServer.call(__MODULE__, {:update, status})
```

Comment:
is strong consistency relevant? a cast would make it more decoupled.

Reply:
I think consistency is good in this case, so I'd rather start with `call` and only `cast` if needed. I'd like the `.Status` to present status of the parent as accurately as possible, and the `update` is trivial.

By `call` we ensure that things like `status.get` will get a consistent result.

Reply:
OK. In this case, BlockGetter is now doing *more* work than it did before (it's waiting for the state update of the child process on several occasions) and blocking his own internal work.

Reply:
but this is not about reducing BlockGetter's work, quite the contrary - this is about having BlockGetter be the unresponsive heavy lifter and put its queryable state somewhere else for responsiveness.

Also as I said, BlockGetter would wait on the super-responsive `.Status` slave, to do trivial work so that's not much blocking.